Python/symtable.c

```diff
@@ -1339,6 +1340,8 @@ symtable_enter_block(struct symtable *st, identifier name, _Py_block_ty block,
      */
     if (prev) {
         ste->ste_comp_iter_expr = prev->ste_comp_iter_expr;
+        /* inherit already set flags */
+        ste->ste_extra_flags = prev->ste_extra_flags;
```

Comment:
They shouldn't be inherited. E.g., this is fine:

```
>>> def f[T: (lambda: (yield))](): pass
... 
```

Reply:
Oh yes. Actually I found an issue with my construction so I'm fixing it now (I'll go with the `char*` alternative because it would simplify the flow).